### PR TITLE
Feat/set&change email

### PR DESCRIPTION
**Pull request by pitb2022**

### CODE REVIEW

File: src/views/Callback/UI.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const UI = ({ hasError }: Props) => {
+  return (
+    <section className={styles.callback}>
+      {hasError && (
+        <section className={styles.error}>
+          {withIcon(IconIllustation)({})}
+          <section className={styles.hint}>
+            <section className={styles.title}>
+              <FormattedMessage
+                defaultMessage="Oops！This link has expired"
```
Comment (robertu7) [inline src/views/Callback/UI.tsx:23]: there are other types of errors should be handle?

File: src/components/Dialogs/SetEmailDialog/Content.tsx [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+          variables: {
+            input: {
+              email: values.email,
+              type: VerificationCodeType.EmailVerify,
+              redirectUrl,
+            },
+          },
+          refetchQueries: [
+            {
+              query: ROOT_QUERY_PRIVATE,
```
Comment (robertu7) [inline src/components/Dialogs/SetEmailDialog/Content.tsx:113]: where do we need to refetch `ROOT_QUERY_PRIVATE` after sending the verification code?

File: src/components/GlobalToasts/EmailVerificationToast/ResendAction.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      variables: {
+        input: {
+          email,
+          type: VerificationCodeType.EmailVerify,
+          redirectUrl,
+        },
+      },
+      refetchQueries: [
+        {
+          query: ROOT_QUERY_PRIVATE,
```
Comment (robertu7) [inline src/components/GlobalToasts/EmailVerificationToast/ResendAction.tsx:69]: same question as above about the refeching.

**Review by robertu7 — APPROVED** [review]

File: src/views/Callback/UI.tsx (hunk 1 above)
Comment (pitb2022) [reply src/views/Callback/UI.tsx:23]: This is the only one for now.

File: src/components/Dialogs/SetEmailDialog/Content.tsx (hunk 2 above)
Comment (pitb2022) [reply src/components/Dialogs/SetEmailDialog/Content.tsx:113]: User.info.email, User.info.emailVerified and User.status.hasEmailLoginPassword need to be updated.

Comment (robertu7) [reply src/components/Dialogs/SetEmailDialog/Content.tsx:113]: These fields won't be updated on `sendCode` mutation is call.

File: src/views/Callback/UI.tsx (hunk 1 above)
Comment (robertu7) [reply src/views/Callback/UI.tsx:23]: https://github.com/thematters/matters-web/pull/3791/files/e1f8e252bf70dfecb99febb9c171e119146147b2#diff-abc78b7a78d045ea04c4a195fb9a4b119c2861658a9d27adbf0e1460598df79aR42 errors from `verifyEmail` should be more than one.

File: src/components/GlobalToasts/EmailVerificationToast/ResendAction.tsx (hunk 3 above)
Comment (pitb2022) [reply src/components/GlobalToasts/EmailVerificationToast/ResendAction.tsx]: removed

File: src/components/Dialogs/SetEmailDialog/Content.tsx (hunk 2 above)
Comment (pitb2022) [reply src/components/Dialogs/SetEmailDialog/Content.tsx]: has updated after SET_EMAIL mutation

**Pull request merged**